modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/Cookie.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+const CONSENT_TYPES = {
+	FUNCTIONAL: 'CONSENT_TYPE_FUNCTIONAL',
+	NECESSARY: 'CONSENT_TYPE_NECESSARY',
+	PERFORMANCE: 'CONSENT_TYPE_PERFORMANCE',
+	PERSONALIZATION: 'CONSENT_TYPE_PERSONALIZATION',
+};
+
+const generateCookie = (name, value, options = {}) =>
+	`${name}=${value}; ${Object.entries(options)
+		.map(([key, value]) => {
+			if (key === 'secure') {
+				return value ? 'secure' : '';
+			}
+
+			return `${key}=${value}`;
+		})
+		.join('; ')}`.trim();
```

Comment:
This function looks kinda complex, by that I mean not very readable. Perhaps it would be a good idea to add a couple of lines of code for the sake of clarity.